specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/serviceTags.json

```diff
@@ -0,0 +1,147 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagementClient",
+    "description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2019-04-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/locations/{location}/serviceTags": {
+      "get": {
+        "operationId": "ServiceTags_List",
+        "description": "Gets a list of service tag information resources.",
+        "parameters": [
+          {
+            "name": "location",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The location."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. Returns a list of service tag information resources.",
+            "schema": {
+              "$ref": "#/definitions/ServiceTagsListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get list of service tags": {
+            "$ref": "./examples/ServiceTagsList.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ServiceTagsListResult": {
+      "properties": {
```

Comment:
This is a bit confusing.
The list result contains `name`, `id` and `type` along with an array of `ServiceTagInformation` objects which in turn have the same `name`, `id`, `type` properties?

Reply:
`name`. `id` in `ServiceTagsListResult` are for from which cloud the data is (Public, Chinese, US Government or German Government).
`name` and `id` of `ServiceTagInformation` show which service the inforrmation is about

```json
{
  "name": "public",
  "id": "/subscriptions/subId/providers/Microsoft.Network/serviceTags/public",
  "values": [
    {
      "name": "ApiManagement",
      "id": "ApiManagement"
     }
  ]
}
```

Maybe @chandrasekarsrinivasan could give more context if it's still unclear

Reply:
Thanks for the explanation. I think someone from ARM should sign this off.
Per your explanation it seems like name, id, type must be readonly

Reply:
Per offline thread, these are read only operations, please mark the appropriate properties as readonly

Reply:
Already sent the ARM approval email.